src/jobs/deploy_env.yml

```diff
@@ -61,8 +61,18 @@ parameters:
   helm_timeout:
     type: string
     default: 5m
+  checkout_branch_override:
+    type: string
+    default: ''
 steps:
-  - checkout
+  - when:
+      condition: <<parameters.checkout_branch_override>>
+      steps:
+        - run: git clone -b "<<parameters.checkout_branch_override>>" "$CIRCLE_REPOSITORY_URL"
+  - unless:
+      condition: <<parameters.checkout_branch_override>>
```

Comment:
Probably missing my understanding but aren't the conditions for the `when` and `unless` the same?